Barotrauma/BarotraumaShared/SharedSource/Characters/Character.cs

```diff
@@ -1358,6 +1358,41 @@ public void GiveJobItems(WayPoint spawnPoint = null)
             info.Job.GiveJobItems(this, spawnPoint);
         }
 
+        // Replace this character with another one at the start of the round or after the respawn shuttle is spawned.
+        public IEnumerable<object> ReplaceWithMonsterJob(Client client, Dictionary<string, bool> characters)
+        {
+#if SERVER
+            string characteridentifier = characters.Keys.GetRandom(Rand.RandSync.Server);
```

Comment:
Perhaps unintuitively, `RandSync.Server` means "synchronized between the server and the clients", which means that if you're going to use this random number generator, the clients also have to do it in the same order and in the same amount. If this works just fine as server-only code, use `RandSync.Unsynced` instead.